src/factories/store/factoriesSelectors.ts

```diff
@@ -35,6 +35,24 @@ export const useFactorySimpleAttributes = (id: string | null | undefined) => {
       id: factory?.id,
       name: factory?.name,
       description: factory?.description,
+      progress: factory?.progress,
+      boardIndex: factory?.boardIndex,
+    };
+  });
+};
+
+export const useSafeFactorySimpleAttributes = (id?: string) => {
+  return useShallowStore(state => {
+    if (!id) {
+      return undefined;
+    }
+    const factory = state.factories.factories[id];
```

Comment:
I'd still return undefined if factory is undefined - it should never happen, but it could be we're accessing a factory which is now deleted, etc., many things we can't really control; 
this shouldn't introduce new behaviour since we're already returning undefined here, I'd add it

Reply:
My experience has shown me that failing early is better than failing later. As this will create harder to debug issues, are harder to maintain code. Thus I tend to prefer actually handling explicitly each case I know of, and once I actually encounter an unexpected case I add it - with a test case that also reproduces it. Anyway this method is not being used so I will just remove it.